base_layer/core/src/services/encryption_service.rs

```diff
@@ -0,0 +1,72 @@
+//  Copyright 2022, The Tari Project
+//
+//  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+//  following conditions are met:
+//
+//  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+//  disclaimer.
+//
+//  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+//  following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+//  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+//  products derived from this software without specific prior written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+//  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+//  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+//  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+//  USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use tari_common_types::types::{Commitment, PrivateKey};
+use tari_service_framework::{async_trait, ServiceInitializationError, ServiceInitializer, ServiceInitializerContext};
+
+use crate::transactions::{tari_amount::MicroTari, transaction_components::EncryptedValue};
+
+// use thiserror::Error;
+//
+// #[derive(Debug, Error)]
+// pub enum EncryptionServiceError {}
+
+// TODO: Rename to `EncryptionFactory`
+#[derive(Debug, Clone)]
+pub struct EncryptionServiceHandle {}
+
+impl EncryptionServiceHandle {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn encrypt_value(&self, key: &PrivateKey, commitment: &Commitment, value: MicroTari) -> EncryptedValue {
+        const SIZE: usize = 24;
+        let mut data: [u8; SIZE] = [0; SIZE];
+        let value = u64::from(value).to_le_bytes();
+        data[0..8].copy_from_slice(&value);
+        EncryptedValue(data)
+    }
+
+    pub fn decrypt_value(&self, value: &EncryptedValue) -> MicroTari {
```

Comment:
This method should throw an error if the value cannot be decrypted, as per Aaron's scanning poc